lib/realtime/tenants/authorization.ex

```diff
@@ -48,31 +51,78 @@ defmodule Realtime.Tenants.Authorization do
   end
 
   @doc """
-  Runs validations based on RLS policies to set policies for a given connection (either Phoenix.Socket or Plug.Conn).
+  Runs validations based on RLS policies to set policies for read policies a given connection (either Phoenix.Socket or Plug.Conn).
   """
-  @spec get_authorizations(Phoenix.Socket.t() | Plug.Conn.t(), DBConnection.t(), __MODULE__.t()) ::
-          {:ok, Phoenix.Socket.t() | Plug.Conn.t()} | {:error, any()}
-  def get_authorizations(%Phoenix.Socket{} = socket, db_conn, authorization_context) do
-    case get_authorizations(db_conn, authorization_context) do
-      {:ok, %Policies{} = policies} -> {:ok, Phoenix.Socket.assign(socket, :policies, policies)}
+  @spec get_read_authorizations(
+          Socket.t() | Conn.t(),
+          DBConnection.t(),
+          __MODULE__.t(),
+          Policies.t()
+        ) ::
+          {:ok, Socket.t() | Conn.t()} | {:error, any()}
+  def get_read_authorizations(socket, db_conn, authorization_context, policies \\ %Policies{})
+
+  def get_read_authorizations(
+        %Socket{} = socket,
+        db_conn,
+        authorization_context,
+        policies
+      ) do
+    policies = Map.get(socket.assigns, :policies, %Policies{}) || policies
+
+    case get_read_policies_for_connection(db_conn, authorization_context, policies) do
+      {:ok, %Policies{} = policies} -> {:ok, Socket.assign(socket, :policies, policies)}
       {:error, error} -> {:error, error}
     end
```

Comment:
a little bit shorter variant 
```suggestion
    with {:ok, %Policies{} = policies} <- get_read_policies_for_connection(db_conn, authorization_context, policies) do
      {:ok, Socket.assign(socket, :policies, policies)}
    end
```